utils/gyb_syntax_support/AttributeNodes.py

```diff
@@ -267,19 +263,49 @@
              Child('Label', kind='IdentifierToken',
                    text_choices=['jvp', 'vjp']),
              Child('Colon', kind='ColonToken'),
-             Child('DeclBaseName', kind='Syntax', description='''
+             Child('FunctionDeclName', kind='FunctionDeclName',
+                   description='The referenced function name.'),
+             Child('TrailingComma', kind='CommaToken', is_optional=True),
+         ]),
+
+    # func-decl-name -> (identifier | operator) decl-name-arguments?
+    # NOTE: This is duplicated with `DeclName` above. Change `DeclName`
+    # description and use it if possible.
+    Node('FunctionDeclName', kind='Syntax',
```

Comment:
`FunctionDeclName` is duplicated with the [`DeclName`](https://github.com/apple/swift/blob/706a75007ba6d6569b248210cbf7e900c3df1b03/utils/gyb_syntax_support/AttributeNodes.py#L126) syntax node.
(The extra `SpacedBinaryOperator` node choice is added to support parsing `-`, which gets parsed as a `SpacedBinaryOperator` and not a `PrefixOperator`).

The reservations for not using `DeclName` are:
* The descriptions for `DeclName` are too specific and need to be made more general. Currently, they're specialized for protocol requirements, e.g. `"The base name of the protocol's requirement."`
* We need to add a `SpacedBinaryOperator` node choice.

@nkcsgexi: is it okay to edit `DeclName` (change descriptions and add node choice), or is that breaking?

Reply:
I'll proceed with the duplicated `FunctionDeclName` syntax node for now.